### PR TITLE
Community Structure

### PR DESCRIPTION
**Pull request by egolol**

### CODE REVIEW

**Review by c-yuwei — APPROVED** [review]

File: braph2genesis/measures/_CommunityStructure.gen.m [hunk 1]
```diff
@@ -0,0 +1,353 @@
...
+rule (metadata, OPTION) 
+%%%% ¡settings!
+{ 'louvain' 'newman'}
+%%%% ¡default!
+'louvain'
+
+%%% ¡prop! 
+gamma (metadata, SCALAR) 
+%%%% ¡default!
+1
```
Comment (annacanal) [inline braph2genesis/measures/_CommunityStructure.gen.m:51]: Gamma is a PARAMETER, not METADATA

METADATA     Metadata NOT used in the calculation of the results (does not allow callbacks).

File: braph2genesis/measures/_CommunityStructure.gen.m [hunk 2]
```diff
@@ -0,0 +1,353 @@
...
+%%% ¡prop! 
+M0 (metadata, rvector) 
+%%%% ¡default!
+[]
+
+%%% ¡prop! 
+OM (metadata, MATRIX) 
+%%%% ¡default!
+[]
+
```
Comment (annacanal) [inline braph2genesis/measures/_CommunityStructure.gen.m:62]: it think these 2 are also Parameter

File: braph2genesis/measures/_CommunityStructure.gen.m [hunk 3]
```diff
@@ -0,0 +1,353 @@
...
+%%% ¡prop! 
+OM_TYPE (metadata, OPTION) 
+%%%% ¡settings!
+{'modularity' 'potts' 'negative_sym' 'negative_asym'}
+%%%% ¡default!
+'modularity'
+
+%%% ¡prop! 
+quality_function (metadata, SCALAR)
```
Comment (annacanal) [inline braph2genesis/measures/_CommunityStructure.gen.m:71]: this is DATA. already calculated quality_function, which is used to calculate community structure

File: braph2genesis/measures/_CommunityStructure.gen.m (hunk 2 above)
Comment (annacanal) [reply braph2genesis/measures/_CommunityStructure.gen.m:62]: actually I think that OM (the B matrix) and the initial community affiliation (M0) are DATA

**Review by annacanal — CHANGES_REQUESTED** [review]

File: braph2genesis/measures/_CommunityStructure.gen.m [hunk 4]
```diff
@@ -39,29 +39,29 @@
...
-rule (metadata, OPTION) 
+rule (parameter, OPTION) 
 %%%% ¡settings!
 { 'louvain' 'newman'}
 %%%% ¡default!
 'louvain'
 
 %%% ¡prop! 
-gamma (metadata, SCALAR) 
+gamma (data, SCALAR)
```
Comment (annacanal) [inline braph2genesis/measures/_CommunityStructure.gen.m:49]: parameter

**Review by annacanal — CHANGES_REQUESTED** [review]

**Review by annacanal — APPROVED** [review]

File: braph2genesis/measures/_CommunityStructure.gen.m [hunk 5]
```diff
@@ -0,0 +1,353 @@
...
+    'modularity' ...
+    'potts' ...
+    'negative_sym' ...
+    'negative_asym'
+    };
+
+%% ¡props!
+%%% ¡prop! 
+rule (parameter, OPTION) 
+%%%% ¡settings!
```
Comment (giovannivolpe) [inline braph2genesis/measures/_CommunityStructure.gen.m:43]: add def

**Review by giovannivolpe — CHANGES_REQUESTED** [review]
Add definitions to props

**Review by giovannivolpe — APPROVED** [review]

**Pull request merged**